server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/FileTimeIndex.java

```diff
@@ -88,23 +82,28 @@ public void close() {
 
   @Override
   public Set<String> getDevices(String tsFilePath, TsFileResource tsFileResource) {
-    FILE_READER_MANAGER.increaseFileReaderReference(tsFileResource, tsFileResource.isClosed());
-    try {
-      TsFileSequenceReader fileReader = FileReaderManager.getInstance().get(tsFilePath, true);
-      return new HashSet<>(fileReader.getAllDevices());
+    try (InputStream inputStream =
+        FSFactoryProducer.getFSFactory()
+            .getBufferedInputStream(tsFilePath + TsFileResource.RESOURCE_SUFFIX)) {
+      // The first byte is VERSION_NUMBER, second byte is timeIndexType.
+      ReadWriteIOUtils.readBytes(inputStream, 2);
+      ITimeIndex timeIndex = new DeviceTimeIndex().deserialize(inputStream);
+      return timeIndex.getDevices(tsFilePath, tsFileResource);
```

Comment:
Better to add a method in DeviceTimeIndex to just get useful info i.e. deviceName. In the new method, you don't need to  do like `ReadWriteIOUtils.readLong(inputStream)`  operations which not only read 8 bytes but also call `BytesUtils.bytesToLong(bytes);` to convert it into long and this is unnecessary.

Reply:
done. `DeviceTimeIndex#getDevices(InputStream inputStream)`